pkg/reconcilers/base_reconciler.go

```diff
@@ -123,48 +165,53 @@ func (b *BaseReconciler) ReconcileResource(ctx context.Context, obj, desired cli
 
 	// item found successfully
 	if common.IsObjectTaggedToDelete(desired) {
-		return b.DeleteResource(ctx, desired)
+		return b.DeleteResource(ctx, desired, reconcileOpts.DeleteOptions...)
 	}
 
-	desired.SetResourceVersion(obj.GetResourceVersion())
-	if err := b.Client().Update(ctx, desired, client.DryRunAll); err != nil {
-		return err
+	var desiredMutated = desired
```

Comment:
Because `desired` is a pointer, the indirection alone would not save from the dry-run func mutating its state, hence the reassignment `desiredMutated = desired.DeepCopyObject()` at line 174, right?

Reply:
the intention is to avoid mutating *desired* object. The caller of the function does not expect a change on it.

Reply:
I was just pointing out that what protects the state of `desired` is calling `DeepCopyObject()` further, and not this line alone.